server/src/domain/metadata/metadata.service.ts

```diff
@@ -381,7 +381,11 @@ export class MetadataService {
       length = videoOffset;
     }
 
-    if (!length) {
+    if (
+      length === 0 &&
+      tags.MotionPhotoVideo === undefined &&
+      (tags.EmbeddedVideoFile === undefined || tags.EmbeddedVideoType !== 'MotionPhoto_Data')
+    ) {
```

Comment:
Can you require this as: not length or no motion tag then return? Maybe it has motion tag as a constant in the line above